src/pkg/services/m365/api/lists.go

```diff
@@ -290,25 +498,67 @@ func CloneListItem(orig models.ListItemable) models.ListItemable {
 func retrieveFieldData(orig models.FieldValueSetable) models.FieldValueSetable {
 	fields := models.NewFieldValueSet()
 	additionalData := make(map[string]any)
-	fieldData := orig.GetAdditionalData()
-
-	// M365 Book keeping values removed during new Item Creation
-	// Removed Values:
-	// -- Prefixes -> @odata.context : absolute path to previous list
-	// .           -> @odata.etag : Embedded link to Prior M365 ID
-	// -- String Match: Read-Only Fields
-	// -> id : previous un
-	for key, value := range fieldData {
-		if strings.HasPrefix(key, "_") || strings.HasPrefix(key, "@") ||
-			key == "Edit" || key == "Created" || key == "Modified" ||
-			strings.Contains(key, "LookupId") || strings.Contains(key, "ChildCount") || strings.Contains(key, "LinkTitle") {
-			continue
-		}
 
-		additionalData[key] = value
+	if orig != nil {
+		fieldData := orig.GetAdditionalData()
+
+		// M365 Book keeping values removed during new Item Creation
+		// Removed Values:
+		// -- Prefixes -> @odata.context : absolute path to previous list
+		// .           -> @odata.etag : Embedded link to Prior M365 ID
+		// -- String Match: Read-Only Fields
+		// -> id : previous un
+		for key, value := range fieldData {
+			_, isReadOnlyField := readOnlyFieldNames[key]
+
+			if strings.HasPrefix(key, "_") ||
+				strings.HasPrefix(key, "@") ||
+				isReadOnlyField ||
+				strings.Contains(key, "LinkTitle") ||
+				strings.Contains(key, "ChildCount") {
+				continue
+			}
+
+			additionalData[key] = value
+		}
 	}
 
+	retainPrimaryAddressField(additionalData)
+
 	fields.SetAdditionalData(additionalData)
 
 	return fields
 }
+
+func retainPrimaryAddressField(additionalData map[string]any) {
+	if !hasAddressFields(additionalData) {
+		return
+	}
+
+	for k := range readonlyAddressFieldNames {
+		delete(additionalData, k)
+	}
+}
+
+func hasAddressFields(additionalData map[string]any) bool {
+	for key, value := range additionalData {
+		if nestedFields, ok := value.(map[string]any); ok &&
+			hasRequiredFields(nestedFields, addressFieldNames) &&
+			hasRequiredFields(additionalData, readonlyAddressFieldNames) &&
+			key != "GeoLoc" {
+			return true
+		}
+	}
+
+	return false
+}
```

Comment:
There's a lot of code packed into this conditional, and it's not very easy to read.  Suggestion for layout:
```suggestion
  nestedFields, ok := value.(map[string]any)
  if !ok || key == "GeoLoc" {
    continue
  }
  if map.HasAllFields(
  
		if ; ok &&
			hasRequiredFields(nestedFields, addressFieldNames) &&
			hasRequiredFields(additionalData, readonlyAddressFieldNames) &&
			key != "GeoLoc" {
			return true
		}
```

```suggestion
func hasAddressFields(additionalData map[string]any) bool {
	if !hasRequiredFields(additionalData, readOnlyAddressFieldNames) {
		return false
	}
	
	for key, value := range additionalData {
		nestedFields, ok := value.(map[string]any)
		if !ok || key == "GeoLoc" {
			continue
		}
		if hasRequiredFields(nestedFields, addressFieldNames) {
			return true
		}
	}

	return false
}
```